runtime/darwinia/src/lib.rs

```diff
@@ -566,14 +566,107 @@ impl_runtime_apis! {
+	216, 45, 91, 24, 197, 246, 158, 52, 170, 240, 72, 212, 249, 52, 107, 173, 97, 36, 136, 64, 183,
+	162, 180, 114, 222, 74, 60, 0, 1, 242, 213, 85, 220, 83, 75, 82, 56, 152, 119, 220, 108, 61,
+	176, 116, 252, 64, 129, 14, 60, 133, 137, 73, 187, 185, 191, 188, 144, 115, 24, 189, 211, 239,
+	59, 14, 142, 28, 91, 120, 34, 22, 121, 27, 168, 237, 99, 152, 44, 219, 209, 130, 135, 33, 15,
+	105, 122, 221, 56, 8, 93, 131, 43, 67, 129,
+];
+
+#[test]
+fn assert_encoded_eq() {
+	darwinia_ethereum_relay::migration::assert_encoded_eq(
+		r#"{
+			"header": {
+				"difficulty": "0x1a76e148d47e3f",
+				"extraData": "0x626565706f6f6c2e6f72675f3622e3",
+				"gasLimit": "0xe4c4d7",
+				"gasUsed": "0xe49a99",
+				"hash": "0xf2d555dc534b52389877dc6c3db074fc40810e3c858949bbb9bfbc907318bdd3",
+				"logsBloom": "0x1226e522b431888c52a186a6bc361a1375930081649a046025492940aec1890d111c0120082c0d1117435383000b4930122689d62c22adc01ad27e8a42e12104b0500454200121bd79d6382ea0c65af40235a904046409b8b44c3602ca883503930483040a270970b566100c2009e940e016522b580d845f4f290e5a2d0d63360800b4c09e002ab632fc12c8028869a2973911011360c5483a26216a1d9803b9aadca1893005a8f36603eac4a942058098300280da0c8b2620e387aa12092a04ec13a063718d56c02aa31330279830048e1d1706150bb97666c877b2023221c7173ae13dd9c10080b964b14d208c64108dc32590e95880500e838a0f22129221",
+				"miner": "0x99c85bb64564d9ef9a99621301f22c9993cb89e3",
+				"mixHash": "0x37690063a5583ebc490c22f1ad1e96d82d5b18c5f69e34aaf048d4f9346bad61",
+				"nonce": "0x40b7a2b472de4a3c",
+				"number": "0xc5814b",
+				"parentHash": "0xaf7ec781faedb3524013da0561afef52aeae79f8f24b5a064d8b6e55c3d54587",
+				"receiptsRoot": "0x789dc0e01565ea1e4291c2b5bdd5e930aebc83bc3ab4646bb78d65fe3daad484",
+				"sha3Uncles": "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
+				"size": "0x1092c",
+				"stateRoot": "0x2836918ac7e0f151d5594fc3276cf6c34cd7b6512d622e72fa393daff72752e1",
+				"timestamp": "0x61076b22",
+				"totalDifficulty": "0x5ffec465fc78e1f3319",
+				"transactions": [],
+				"transactionsRoot": "0xdca3e219bda6af76807a78ee412054aba59acc347b209d8db55e49836beb8d50",
+				"uncles": []
+			},
+			"parent_mmr_root": "0xef3b0e8e1c5b782216791ba8ed63982cdbd18287210f697add38085d832b4381"
+		}"#,
+		PARCEL,
+	)
+}
+
 pub struct CustomOnRuntimeUpgrade;
 impl OnRuntimeUpgrade for CustomOnRuntimeUpgrade {
 	#[cfg(feature = "try-runtime")]
 	fn pre_upgrade() -> Result<(), &'static str> {
+		// --- paritytech ---
+		use frame_support::{migration, Identity};
+		// --- darwinia-network ---
+		use darwinia_header_mmr::NodeIndex;
+
+		darwinia_header_mmr::migration::migrate(b"HeaderMMR");
+
+		assert!(migration::storage_key_iter::<NodeIndex, Hash, Identity>(
+			b"HeaderMMR",
+			b"MMRNodeList"
+		)
+		.next()
+		.is_none());
+		assert!(!migration::have_storage_value(
+			b"HeaderMMR",
+			b"MMRNodeList",
+			&[]
+		));
+		assert!(!migration::have_storage_value(
+			b"HeaderMMR",
+			b"PruningConfiguration",
+			&[]
+		));
```

Comment:
Have we already remove `PruningConfiguration` in previous migrations?

Reply:
Nope, you can still query it from chain state.